sdks/csharp/sdk/AgonesSDK.cs

```diff
@@ -174,24 +187,42 @@ public async Task<GameServer> GetGameServerAsync()
 		/// Starts watching the GameServer updates in the background in it's own task.
 		/// On update, it fires the GameServerUpdate event.
 		/// </summary>
-		private async Task BeginInternalWatch()
+		private async Task BeginInternalWatchAsync()
 		{
-			//Begin WatchGameServer in the background for the provided callback.
-			try
+			// Begin WatchGameServer in the background for the provided callback(s).
+			while (!ctoken.IsCancellationRequested)
 			{
-				using (watchStreamingCall = client.WatchGameServer(new Empty(), cancellationToken: ctoken))
+				try
 				{
-					var reader = watchStreamingCall.ResponseStream;
-					while (await reader.MoveNext(ctoken))
+					using (var watchStreamingCall = client.WatchGameServer(new Empty(), cancellationToken: ctoken))
 					{
-						GameServerUpdated?.Invoke(reader.Current);
+						var reader = watchStreamingCall.ResponseStream;
+						while (await reader.MoveNext(ctoken))
+						{
+							try
+							{
+								GameServerUpdated?.Invoke(reader.Current);
+							}
+							catch
+							{
+								// Ignore any exception thrown here. We don't want a callback's exception to cause
+								// our watch to be torn down.
+							}

```

Comment:
I tend to be wary of empty catches. Perhaps you would consider adding something, even if it is a debug log?

Reply:
Fair, I'll add a log